apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -1503,4 +1828,195 @@ private void writeDebugLogMessages() {
+      annotations.add(idAnnotation.build());
+    }
+    // Add an @Column annotation to every non-transient column.
+    if (!isTransient) {
+      AnnotationSpec.Builder columnAnnotation =
+          AnnotationSpec.builder(Column.class)
+              .addMember("name", "$S", "`" + fieldName + "`")
+              .addMember("nullable", "$L", isColumnOptional);
+      if (type == RifColumnType.CHAR && columnLength.isPresent()) {
+        columnAnnotation.addMember("length", "$L", columnLength.get());
+      } else if (type == RifColumnType.NUM) {
+        /*
+         * In SQL, the precision is the number of digits in the unscaled value, e.g.
+         * "123.45" has a precision of 5. The scale is the number of digits to the right
+         * of the decimal point, e.g. "123.45" has a scale of 2.
+         */
+
+        if (columnLength.isPresent() && columnScale.isPresent()) {
+          columnAnnotation.addMember("precision", "$L", columnLength.get());
+          columnAnnotation.addMember("scale", "$L", columnScale.get());
+        } else {
+          /*
+           * Unfortunately, Hibernate's SQL schema generation (HBM2DDL) doesn't correctly
+           * handle SQL numeric datatypes that don't have a defined precision and scale.
+           * What it _should_ do is represent those types in PostgreSQL as a "NUMERIC",
+           * but what it does instead is insert a default precision and scale as
+           * "NUMBER(19, 2)". The only way to force the correct behavior is to specify a
+           * columnDefinition, so we do that. This leads to incorrect behavior with HSQL
+           * (for different reasons), but fortunately that doesn't happen to cause
+           * problems with our tests.
+           */
+          StringBuilder columnDefinition = new StringBuilder();
+          columnDefinition.append("numeric");
+          if (columnLength.isPresent() || columnScale.isPresent()) {
+            columnDefinition.append('(');
+            if (columnLength.isPresent()) {
+              columnDefinition.append(columnLength.get());
+            }
+            if (columnScale.isPresent()) {
+              columnDefinition.append(", ");
+              columnDefinition.append(columnScale.get());
+            }
+            columnDefinition.append(')');
+          }
+          columnAnnotation.addMember("columnDefinition", "$S", columnDefinition.toString());
+        }
+      }
+      annotations.add(columnAnnotation.build());
+    } else {
+      annotations.add(AnnotationSpec.builder(Transient.class).build());
+    }
+
+    return annotations;
+  }
+
+  /**
+   * Selects the java field type
+   *
+   * @param type specifies the field type {@link RifColumnType}
+   * @param isColumnOptional determines if the field is optional {@link boolean}
+   * @param columnLength specifies the column length {@link Optional<Integer>}
+   * @param columnScale specifies the column scale {@link Optional<Integer>}
```

Comment:
Would you mind applying this suggestion?
```suggestion
   * @param columnLength specifies the column length {@link Optional<Integer>}, for numeric types this represents the total number of digits that can be stored
   * @param columnScale specifies the column scale {@link Optional<Integer>}, for numeric types this represents how many of the total digits (see `columnLength`) are to the right of the decimal point
```